### DISCUSSION TITLE
What "bailout" functionality is available in pest to parse non-context-free syntax?

### DISCUSSION THREAD

**Discussion by ahelwer** [pest-parser/pest#1011] [Q&A] [answered]
For example, tree-sitter has an external scanner written in C to parse the non-context-free part of a language while using the grammar for everything else. Is something similar available for pest?

**Comment by tomtau** [comment, marked as answer]
There's no general escape hatch, but there are language constructs where it can parse expressions that go beyond CFG: see https://pest.rs/book/grammars/syntax.html#the-stack-wip

**Reply by ahelwer** [reply]
Thanks. Unfortunately I don't think stacks will fully work to parse TLA+, because the vertically-aligned blocks often aren't preceded by whitespace on the line. For example, this nested conjunction list:
```tla
op ==
  /\ x
  /\ /\ y
     /\ y
  /\ x
```
Is there a way to access the column position (in codepoints) inside the PEG and push that to the stack somehow then match on it?

@beggers as FYI

**Reply by tomtau** [reply]
not sure if this works: https://pest.rs/book/grammars/syntax.html#indentation-sensitive-languages ?

**Reply by ahelwer** [reply]
It does not, since the preceding space in the line is not whitespace.
